### PR TITLE
Added doc for Catalan number

### LINKED ISSUE #88: Documentation for Catalan number
## 📚 Documentation

Documentation for Catalan number.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

Please assign this to me under GSSOC'21.

### PR DESCRIPTION
**Pull request by nidh-ish**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added doc for Catalan number.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #88

### CODE REVIEW

File: docs/mathematics/catalan-number.mdx [hunk 1]
```diff
@@ -0,0 +1,183 @@
+---
+title: Catalan number
+slug: /mathematics/catalan-number
+---
+
+## Introduction
+
+Catalan numbers are a sequence of natural numbers that occur in various counting problems. The solution to many counting problems in combinatorial mathematics is given by Catalan numbers.
```
Comment (HarshCasper) [inline docs/mathematics/catalan-number.mdx:8]: Can you share a bit more description? It seems too vague to me.

File: docs/mathematics/catalan-number.mdx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+n<sup>th</sup> Catalan number for `n >= 0`:
+
+<a href="https://www.codecogs.com/eqnedit.php?latex=C_{n&plus;1}&space;=&space;\sum_{i=0}^n{C_iC_{n-i}}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?C_{n&plus;1}&space;=&space;\sum_{i=0}^n{C_iC_{n-i}}" title="C_{n+1} = \sum_{i=0}^n{C_iC_{n-i}}" /></a>
+
+## Algorithm
+
+#### Recursive
+
+```
```
Comment (HarshCasper) [inline docs/mathematics/catalan-number.mdx:30]: Can you add some text based instructions as well?

File: docs/mathematics/catalan-number.mdx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+for num = 0 to n
+   catalan += catalan(num) + catalan(n - num - 1)
+return catalan
+```
+
+This takes `exponential` running time.
+
+#### Dynamic Programming
+
+```
```
Comment (HarshCasper) [inline docs/mathematics/catalan-number.mdx:41]: Can you add some text based instructions as well?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/mathematics/catalan-number.mdx (hunk 1 above)
Comment (nidh-ish) [reply docs/mathematics/catalan-number.mdx:8]: I searched it online. There are its applications, but nothing more to explain.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**